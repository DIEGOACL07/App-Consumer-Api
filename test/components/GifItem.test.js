import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas en <GifItem />', () => {

    const title = 'Diego';
    const url = 'https://one-punch.com/saitama.jpg';

    test('debe de hacer match con el snapchot', () => {
        
        const {container } = render(<GifItem title={title}  url={url}></GifItem>)
        expect( container ).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con el url indicado', () => {
      
        render(<GifItem title={title}  url={url}></GifItem>)
        //screen.debug();
        //expect( screen.getByRole('img').src).toBe( url );

        const {src, alt } = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt );
    })

    test('debe de mostrar el titulo en el componente', () => {
        
        render(<GifItem title={title}  url={url}></GifItem>)
        expect( screen.getByText( title ) ).toBeTruthy();
    });
    
    
    
});
