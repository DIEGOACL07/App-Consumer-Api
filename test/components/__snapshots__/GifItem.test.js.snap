// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe de hacer match con el snapchot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Diego"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      Diego
    </p>
  </div>
</div>
`;
